File: Leetcode/DP/Medium/UniquePaths.py
class Solution(object):
    def uniquePaths(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        dp = [] 
        for i in range(m):          # 申请一个m*n的矩阵，并将值初始化为1
            dp.append([1]*n)

        for i in range(1, m):         # 从1开始遍历，得到每一个位置的路径树
            for j in range(1, n):
                dp[i][j] = dp[i-1][j] + dp[i][j-1]
        
        return dp[m-1][n-1]       # 最终结果

if __name__ == '__main__':
	solu = Solution();
	print(solu.uniquePaths(3,2))
	print(solu.uniquePaths(23,12))

File: Leetcode/BackTrack/Medium/UniquePaths.cc

#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int uniquePaths(int m, int n) {
  		int count = 0;
  		func(1,1,n,m,count);
  		return count;
    }

    void func(int right, int down, int rows, int cols, int &count){
    	if(down==rows && right==cols)
    		count++;
    	if(down>rows || right>cols)
    		return;

    	if(right!=cols)
    		func(right+1,down,rows,cols,count);
    	if(down!=rows)
    		func(right,down+1, rows,cols,count);
    }
};

int main(int argc, char const *argv[])
{
	Solution solu;
	cout<<solu.uniquePaths(3,2)<<endl;
	cout<<solu.uniquePaths(23,12)<<endl;
	return 0;
}